### PR TITLE
[jit] Support get/setstate with no args

### PR DESCRIPTION
**Pull request by driazati**
`pickle` supports this and a lot of the quantized use cases for get/set
state follow this pattern

Differential Revision: [D16391234](https://our.internmc.facebook.com/intern/diff/16391234/)

### CODE REVIEW

File: torch/csrc/jit/export.cpp [hunk 1]
```diff
@@ -751,12 +751,19 @@ bool ScriptModuleSerializer::moduleHasValidGetSetState(
...
-      " arguments");
+  if (set_schema.arguments().size() == 1) {
+    TORCH_CHECK(
+        get_schema.returns().at(0).type()->isSubtypeOf(NoneType::get()),
+        "For '__setstate__' to have no state"
+        " param, '__getstate__' must return None");
+  } else {
+    TORCH_CHECK(
+        set_schema.arguments().size() == 2,
+        "'__setstate__' must have 'self' and the state as its "
```
Comment (eellison) [inline torch/csrc/jit/export.cpp]: you don't check that `self` is the first argument here, maybe that's checked elsewhere

**Review by eellison — APPROVED** [review]
LGTM

File: torch/csrc/jit/export.cpp (hunk 1 above)
Comment (driazati) [reply torch/csrc/jit/export.cpp]: It's a only ever a method so `self` is already installed as the first argument

**Comment by zdevito** [comment]
Where are you getting these semantics from? According to the documentation, and to a test I just did with pickle, if `__getstate__` returns None or False, then `__setstate__` is simply not called. https://docs.python.org/3/library/pickle.html#object.__setstate__

**Pull request closed without merging**